### PR TITLE
Bump client-go manually to fix build issues

### PR DESCRIPTION
**Pull request by ericvn**
**Please provide a description of this PR:**

Bump client-go manually to fix build issues

### CODE REVIEW

**Comment by ericvn** [comment]
See https://prow.istio.io/job-history/gs/istio-prow/logs/update-client-go-dep_client-go_postsubmit

**Review by jacob-delgado — APPROVED** [review]

**Review by keithmattix — APPROVED** [review]

**Comment by ericvn** [comment]
@howardjohn I thought we only were making **required** fields required so we must just have tests that didn't have them set. Does this mean that customer might not have been having **required** fields set? At least at quick glance at test results, it seems we are missing some **required** fields.

**Review by hanxiaop — APPROVED** [review]

**Review by hzxuzhonghu — APPROVED** [review]

File: pilot/pkg/model/sidecar_test.go [hunk 1]
```diff
@@ -165,7 +165,7 @@ var (
 		Spec: &networking.Sidecar{
 			Egress: []*networking.IstioEgressListener{
 				{
-					Port: &networking.Port{
+					Port: &networking.SidecarPort{
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/sidecar_test.go:168]: Isn't this a breaking change

**Review by howardjohn — APPROVED** [review]

**Comment by howardjohn** [comment]
The integration tests fail because they are rejected *for a different reason*. This is WAI, we just need to make the test better (although long term I intend to make it impossible to hit the webhook long term...)

**Comment by hanxiaop** [comment]
It looks like this needs to be fixed before the branch cut step 1 - Update Dependencies.

**Comment by hanxiaop** [comment]
I'm trying to fix the tests in https://github.com/istio/istio/pull/47364, since the client-go dependency blocks the 1.20 branch cut.

**Comment by ericvn** [comment]
Superseded by https://github.com/istio/istio/pull/47377

**Pull request closed without merging**